### PR TITLE
Reject batch change [updated]

### PR DESCRIPTION
**Pull request by britneywright**
rebase of #693 while @mitruly is on vacation 🍹😎

### CODE REVIEW

File: modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala [hunk 1]
```diff
@@ -432,4 +436,25 @@ class BatchChangeService(
         ignoreAccess = ignoreAccess)
     } yield listWithGroupNames
   }
+
+  def rejectBatchChange(
+      batchChange: BatchChange,
+      reviewComment: Option[String],
+      reviewerId: String): BatchResult[BatchChange] = {
+    val rejectedSingleChanges = batchChange.changes.map {
+      case sad: SingleAddChange => sad.copy(status = SingleChangeStatus.Rejected)
```
Comment (pauljamescleary) [inline modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala]: I don't think you need the pattern match here.

```scala
val rejectedChanges = batchChange.changes.map(_.copy(status = SingleChangeStatus.Rejected))
```

Comment (britneywright) [reply modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala]: actually that results in an error ` value copy is not a member of vinyldns.core.domain.batch.SingleChange`

Comment (pauljamescleary) [reply modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala]: Ugh, that is annoying.  Ok.

File: modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala [hunk 2]
```diff
@@ -432,4 +436,25 @@ class BatchChangeService(
...
+      case sdc: SingleDeleteChange => sdc.copy(status = SingleChangeStatus.Rejected)
```
Comment (pauljamescleary) [inline modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala]: Could this be on the `SingleChange` trait itself?

```
sealed trait SingleChange {
  def reject: SingleChange = this match { 
    case sad: SingleAddChange => sad.copy(status = SingleChangeStatus.Rejected)
    case sdc: SingleDeleteChange => sdc.copy(status = SingleChangeStatus.Rejected)
  }
}

...

val rejectedSingleChanges = batchChange.changes.map(_.reject)
```

I also wonder if it is better just to have a `reject` on the `BatchChange` itself that takes care of this?

**Pull request merged**